alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -345,21 +345,30 @@ valContext ::
   Data era
 valContext txinfo sp = Data (P.toData (P.ScriptContext txinfo (transScriptPurpose sp)))
 
+data ScriptResult = Passes | Fails ![String]
+
+andResult :: ScriptResult -> ScriptResult -> ScriptResult
```

Comment:
Could just add a `Semigroup` instance instead. Then we get other combining functions for free

Reply:
good idea, done.